pkg/generic/httpthrift_codec.go

```diff
@@ -94,11 +116,20 @@ func (c *httpThriftCodec) Marshal(ctx context.Context, msg remote.Message, out r
 
 	inner := thrift.NewWriteHTTPRequest(svcDsc)
 	inner.SetBinaryWithBase64(c.binaryWithBase64)
+	inner.SetEnableDynamicgo(c.enableDynamicgoReq)
+	if c.enableDynamicgoReq {
+		inner.SetConvOptions(c.convOpts)
+		inner.SetMethodName(msg.RPCInfo().Invocation().MethodName())
```

Comment:
I think this MethodName() should be empty because user didn't pass method name when generic call. You should Lookup the specific method with the router by req.

Reply:
Sorry for express a wrong statement, we can use MethodName() to get the method name because we've got it in client/genericclient/client.go L98.